lib/scheduler.registry.ts

```diff
@@ -1,12 +1,15 @@
 import { Injectable, Logger } from '@nestjs/common';
-import { CronJob } from 'cron';
+import type { CronJob, CronOnCompleteCommand } from 'cron';
 import { DUPLICATE_SCHEDULER, NO_SCHEDULER_FOUND } from './schedule.messages';
 
 @Injectable()
-export class SchedulerRegistry {
+export class SchedulerRegistry<
+  OC extends CronOnCompleteCommand | null = null,
```

Comment:
I don't think the entire SchedulerRegistry class should be generic. It would mean we can only add/retrieve a single subtype of a job from the registry.

Rather, only the `addCronJob` method itself should be generic, because we can add all kinds of jobs using a single registry.